test/unit/org/apache/cassandra/cql3/validation/miscellaneous/SSTablesIteratedTest.java

```diff
@@ -108,10 +109,10 @@ public void testSinglePartitionQuery() throws Throwable
         executeAndCheck("SELECT * FROM %s WHERE pk = 2 AND c > 20 ORDER BY c DESC", 2,
                         row(2, 40, "42"));
 
-        // Test with only 2 of the 3 SSTables being merged and a Name filter
+        // Test with only 1 of the 3 SSTables being merged and a Name filter
         // This test checks the SinglePartitionReadCommand::queryMemtableAndSSTablesInTimestampOrder which is only
         // used for ClusteringIndexNamesFilter when there are no multi-cell columns
-        executeAndCheck("SELECT * FROM %s WHERE pk = 2 AND c = 10", 2,
+        executeAndCheck("SELECT * FROM %s WHERE pk = 2 AND c = 10", 1,
```

Comment:
It doesn't look like there was previously any logical reason to hit more than the latest SSTable here. The update completely specifies the row.